### PR TITLE
resolves #15. replace lodash with ESX features

### LINKED ISSUE #15: Replace lodash with ESX equivalents
Typically I avoid bring up the tired discussion of "you might not need underscore," but this project boasts being "written in pure ES6." That being said, when searching through some of the source code I discovered some basic usages of lodash. Curiosity got the best of me so I grepped this directory for usages of lodash and found the following unique cases

```
_.extend
_.filter
_.isArray
_.isObject
_.isString
_.map
_.remove
_.toArray
```

Usages of extend, filter, isArray, map, remove, and toArray can be replaced by built-in features of JavaScript (some much older than ES6). The others (isString and isObject) have simple, one line equivalents. Are you opposed to removing lodash and using the builtin features? If you're okay with it, I'd can submit a PR.

### PR DESCRIPTION
**Pull request by blockloop**

### CODE REVIEW

**Comment by dthree** [comment]
Amazing, and looks great!!

Thank you! :+1:

File: src/commands/alias.js [hunk 1]
```diff
@@ -64,9 +62,8 @@ const alias = {
     } else {
       if (value) {
         vorpal.localStorage.setItem(`alias|${key}`, value);
-        _.remove(all, function (val) {
-          return val === key;
-        });
+        const index = all.indexOf(key);
```
Comment (dthree) [inline src/commands/alias.js]: Whoops @blockloop 

On future PRs, make sure you run `gulp` first to build (sorry didn't catch that). This broke alias:

```js
const index = all.indexOf(key);
all.splice(index, 1);
```

**Comment by blockloop** [comment]
Can that be made part of `npm test`?

File: src/commands/alias.js (hunk 1 above)
Comment (blockloop) [inline src/commands/alias.js]: Do I need to fix something or did running Gulp fix the problem?

**Comment by dthree** [comment]
Yes. I'll add it. And don't worry - fixed it! :sunglasses:

**Pull request merged**